tools/releaser/scripts/update-version.sh

```diff
@@ -12,9 +12,14 @@ echo "Current Resource Version: $NEW_RESOURCE_VERSION"
 
 major_version_breaking_changes="${SDK_MAJOR_VERSION: -3}"
 # Remove leading zeros from the variable
-major_version_breaking_changes=${major_version_breaking_changes#00}
-major_version_bump=$((major_version_breaking_changes+1))
-major_version_bump=$(printf "%03d" "$major_version_bump")
+major_version_breaking_changes=$((10#$major_version_breaking_changes))
+echo $major_version_breaking_changes "major_version_breaking_changes"
+# Change value
+((major_version_breaking_changes++))
+# Ensure the value stays within the range 1-999
+major_version_breaking_changes=$((major_version_breaking_changes % 1000))
+
+major_version_bump=$(printf "%03d" "$major_version_breaking_changes")
```

Comment:
In retrospect Bash is not great env for any string/math operation.
Logic above were removing 00 leading zeros from string. We needed to remove any leading ones.